content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,519 @@
+
+df['petal_width_new'] = df.apply(petal_cat, axis=1)  
+```
+The non-machine-friendly format for our *species* class has to be transformed to 0,1 and 2, which is a machine-readable format for the testing to be carried out. Therefore, with [label-encoding](https://www.section.io/engineering-education/introduction-to-scikit-learn-in-python/), a numeric value will be assigned to each of the categories in the class. To learn more about label encoding, [click](https://www.section.io/engineering-education/data-preprocessing-in-r/).
+
+```python
+def species_cat(df):
+ if df["species"] == "Iris-virginica":
+ return 0
+ elif df["species"] == "Iris-versicolor":
+ return 1
+ else:
+ return 2
+
+df["species"] = df.apply(species_cat, axis=1)
+```
+```python
+df.sample(5)
+```
+![dfsample5](dfsample5.png)
+```python
+df_new = df.drop(columns=["sepal_width", "sepal_length", "petal_length", "petal_width"])
+```
+```python
+df_new.head()
+```
+![newhead](df_new.png)
+
+Let's make use of `chic2_contigency` provided for us in `scipy.stats` library.
+
+It has been added to our dependencies earlier in the notebook.
+```python
+stat, p, dof, expected= chi2_contingency(df_new)
+
+print(dof)
+```
+![dof](DOF.png)
+
+We can comfortably get 3 outputs in test statistics, degree of freedom, p-value, and expected values.
+
+[Test statistic](https://online.stat.psu.edu/statprogram/reviews/statistical-concepts/chi-square-tests) is a characterized feature of significance in the Chi-square test that helps in determining how uncommon the result might be provided that null hypothesis is assumed to be true.
+
+The number of features or variables that we have access to varying is the degree of freedom. Follow the link for more information on [degree of freedom](https://byjus.com/maths/chi-square-test/).
+
+Hence, we set our probability of confidence to 95% as explained earlier in the tutorial.
+```python
+prob = 0.95
+
+alpha = 1.0 - prob
+
+print('alpha/significance level=%.3f' % alpha)
+print('P-value =%.2f' % p)
+
+if p <= alpha:
+ print('Reject the Null Hypothesis ( reject H0)')
+else: 
+ print('Accept the Null Hypothesis (fail to reject H0)')
+```
+![alphsig](alphasiglevel.png)
+
+The P-value is greater than the alpha value set at 0.05, we will fail to reject the null hypothesis(we accept the null hypothesis). This means the petal width and species of flower are not dependent, i.e. there is no relationship between them.
+
+#### T-Test
+
+The T-test is utilized when we plan to evaluate the discrepancy between the means of two groups of samples. Unlike Chi-Square Test, the T-test is used on continuous variables.
+
+It has three basic types
+
+1. **One sample t-test:** It checks whether the population has a different mean from the sample.
+The mathematical formula is:
+```
+t-test = a-BX∕N
+```
+Where;
+
+a = the observed mean group.
+
+B = the assumed population means.
+
+X = the standard deviation of the data group.
+
+N = the number of observations in the group.
+
```

Comment:
Formatting error; a similar instance has been pointed out in previous comments. Do make note of that and make the necessary changes for two-sample and paired t-tests.

Reply:
done